src/components/Header/Header.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>;
+  userId: number;
+  setTodos: (todos: Todo[]) => void;
+  todos: Todo[];
+  setErrorType: (errorType: string) => void;
+};
+
+export const Header: React.FC<Props>
+  = ({
+    newTodoField,
+    userId,
+    setTodos,
+    todos,
+    setErrorType,
+  }) => {
+    const [newTodoTitle, setNewTodoTitle] = useState('');
+
+    const setAllCompleted = () => {
+      setErrorType('');
+      setTodos(todos.map((todo: Todo) => {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }));
+    };
+
+    const handleNewTodoTitleChange
+      = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setNewTodoTitle(event.target.value);
+      };
+
+    const handleNewTodoTitleSubmit = (event: React.FormEvent) => {
+      event.preventDefault();
+
+      if (newTodoTitle === '') {
```

Comment:
You need to add .trim() check - to disable adding of empty title (spaces only). Now I can add title name with spaces only.